apps/pipe-easy/src/app/app.component.ts

```diff
@@ -1,21 +1,21 @@
 import { NgFor } from '@angular/common';
-import { Component } from '@angular/core';
-
-@Component({
+import { Component, Pipe, PipeTransform } from '@angular/core';
+@Pipe({
+  name: 'compute',
+  pure: true,
   standalone: true,
-  imports: [NgFor],
+})
+export class HeavyComputationPipe implements PipeTransform {
+  transform(value: { name: string; index: number }) {
+    return `${value?.name} - ${value?.index}`;
```

Comment:
never really though of passing my value this way, but this is working. Doing this way, if name of index change, both will be recompute. In our case this is what we want, so this is not an issue. But if you have static values you should consider doing like the following: 

the most common way is to have multiple property like 
```typescript
transform(name: string, index: number ) {
```

and in your html, you call your pipe like this:
```html
arg1 | mypipe : arg2
```

Reply:
yeah okay understood @tomalaforge